pandas/_libs/tslibs/parsing.pyx

```diff
@@ -1019,6 +1019,11 @@ def guess_datetime_format(dt_str: str, bint dayfirst=False) -> str | None:
 
             output_format.append(tokens[i])
 
+    # if am/pm token present, replace 24-hour %H, with 12-hour %I
+    if "%p" in format_guess and "%H" in format_guess:
+        i = format_guess.index("%H")
+        format_guess[i] = "%I"
```

Comment:
TBH I'm worried this is going to introduce issues - how about just removing `meridiem` from the guessed tokens and not guessing these formats at all?

Reply:
In that case we should just close the issue?

Reply:
sorry what i meant was to remove the line `(("meridiem",), "%p", 0),` and adjust tests accordingly

Reply:
Ah, my bad, I misunderstood. I'll get to it.